internal/mode/static/telemetry/collector.go

```diff
@@ -263,3 +259,55 @@ func CollectClusterID(ctx context.Context, k8sClient client.Reader) (string, err
 	}
 	return string(kubeNamespace.GetUID()), nil
 }
+
+// CollectNodeList returns a NodeList of all the Nodes in the cluster.
+func CollectNodeList(ctx context.Context, k8sClient client.Reader) (v1.NodeList, error) {
+	var nodes v1.NodeList
+	if err := k8sClient.List(ctx, &nodes); err != nil {
+		return nodes, fmt.Errorf("failed to get NodeList: %w", err)
+	}
+
+	return nodes, nil
+}
+
+type clusterInformation struct {
+	Platform  string
+	Version   string
+	ClusterID string
+	Nodes     v1.NodeList
+}
+
+func collectClusterInformation(ctx context.Context, k8sClient client.Reader) (clusterInformation, error) {
+	var clusterInfo clusterInformation
+
+	nodes, err := CollectNodeList(ctx, k8sClient)
+	if err != nil {
+		return clusterInformation{}, fmt.Errorf("failed to collect cluster information: %w", err)
+	}
+	if len(nodes.Items) == 0 {
+		return clusterInformation{}, errors.New("failed to collect cluster information: NodeList length is zero")
+	}
+
+	clusterInfo.Nodes = nodes
+
+	var clusterID string
+	if clusterID, err = CollectClusterID(ctx, k8sClient); err != nil {
+		return clusterInformation{}, fmt.Errorf("failed to collect cluster information: %w", err)
+	}
+	clusterInfo.ClusterID = clusterID
+
+	node := nodes.Items[0]
+	clusterInfo.Version = node.Status.NodeInfo.KubeletVersion
```

Comment:
do we need to trim everything after the version? For example, `v1.27.6+k3s1` -> `v1.27.6`

Reply:
I added a parsing function. Unsure of where to put it as I unit tested it. Currently the tests are in `collector_test.go`, should I move the function to a new file and create a new test file just for that? 

Also, there are many edge cases that I simply wasn't able to cover without it becoming too complicated. Is what I supplied sufficient?

Reply:
> I added a parsing function. Unsure of where to put it as I unit tested it. Currently the tests are in collector_test.go, should I move the function to a new file and create a new test file just for that?

this seem reasonable

> Also, there are many edge cases that I simply wasn't able to cover without it becoming too complicated. Is what I supplied sufficient?

as long as they cover all code paths.